### PR TITLE
Add unfire event and handler for the same

### PR DESCRIPTION
**Pull request by MReyna12**

### CODE REVIEW

**Comment by MReyna12** [comment]
I am open to suggestions on `problemResultsResetValue` in `ProblemSetBlock.tsx`.

File: src/components/ProblemSetBlock.tsx [hunk 1]
```diff
@@ -99,6 +101,7 @@ export const ProblemSetBlock = ({ waitForEvent, fireSuccessEvent, fireLearningOp
     // are either solved or exhausted
     const solvedCount = Array.from(problemResults.values()).reduce((acc, prob) => (prob.solved ? acc + 1 : acc), 0)
     const exhaustedCount = Array.from(problemResults.values()).reduce((acc, prob) => (prob.exhausted ? acc + 1 : acc), 0)
+    const problemResultsResetValue = Array.from(problemResults.values())[0].reset
```
Comment (rnathuji) [inline src/components/ProblemSetBlock.tsx:104]: This is currently hard coded to only work if the first problem is reset and doesn't work with psets that have multiple problems if you reset any of the rest. I think the following reducer will result in `true` if any of the problems are `true` (but, you should confirm / check my logic :sweat_smile: ):

```javascript
const problemResultsResetValue = Array.from(problemResults.values()).reduce((acc, prob) => (prob.reset ? true : acc), false)
```

**Review by rnathuji — COMMENTED** [review]
The problem set unfire approach seemed to come together nicely with the `resetCallback` addition :tada: . I just noticed one minor issue in the implementation when reviewing.

File: src/components/ProblemSetBlock.tsx (hunk 1 above)
Comment (MReyna12) [reply src/components/ProblemSetBlock.tsx:104]: Your reducer does operate as I would expect it to / results in `true` upon pressing the `reset` button. All of my "test" examples use one problem per pset. I'll adjust then moving forward to hopefully catch that undesired behavior. Thanks for the rewrite!

**Review by rnathuji — APPROVED** [review]

**Pull request merged**